### PR TITLE
Worlds implementation

### PR DESCRIPTION
**Pull request by husseinfahmy**
Was finally able to get the modal to show with the functional worlds component. That's still in progress at the moment.

### CODE REVIEW

**Comment by asmaaaah** [comment]
Few questions:
1. The ice world is not greyed out, and does not have lock on it. Any reason we did not do that?

2.This is more for @rampras3 and @natashadecoste since we don't have images for water, jungle and alien world maybe we should leave them out. Now they are displaying ice world which is misleading. Do need to take out the code though, maybe comment it out for until we have images for that. But maybe not in demo.

**Comment by husseinfahmy** [comment]
Regarding your first point - I didn't want to hide Natasha's pretty graphic hahaha. Maybe we can change the locked worlds to say coming soon? That way we can show the ice world while keeping it locked.

**Comment by rampras3** [comment]
> Few questions:
> 
> 1. The ice world is not greyed out, and does not have lock on it. Any reason we did not do that?
> 
> 2.This is more for @rampras3 and @natashadecoste since we don't have images for water, jungle and alien world maybe we should leave them out. Now they are displaying ice world which is misleading. Do need to take out the code though, maybe comment it out for until we have images for that. But maybe not in demo.

Yes, okay. Originally, I thought the rest of the worlds would be greyed out so it wouldn't matter which image we used ??

**Comment by asmaaaah** [comment]
> > Few questions:
> > 
> > 1. The ice world is not greyed out, and does not have lock on it. Any reason we did not do that?
> > 
> > 2.This is more for @rampras3 and @natashadecoste since we don't have images for water, jungle and alien world maybe we should leave them out. Now they are displaying ice world which is misleading. Do need to take out the code though, maybe comment it out for until we have images for that. But maybe not in demo.
> 
> Yes, okay. Originally, I thought the rest of the worlds would be greyed out so it wouldn't matter which image we used ??

They will be but you still see the igloo and all, you can tell it's the same as ice world. Which is misleading.

**Comment by asmaaaah** [comment]
> Regarding your first point - I didn't want to hide Natasha's pretty graphic hahaha. Maybe we can change the locked worlds to say coming soon? That way we can show the ice world while keeping it locked.

Lol yeah makes total sense, it is very pretty. Btw do we have the functional for it to unlock when the fire world is complete?

**Comment by rampras3** [comment]
> > > Few questions:
> > > 
> > > 1. The ice world is not greyed out, and does not have lock on it. Any reason we did not do that?
> > > 
> > > 2.This is more for @rampras3 and @natashadecoste since we don't have images for water, jungle and alien world maybe we should leave them out. Now they are displaying ice world which is misleading. Do need to take out the code though, maybe comment it out for until we have images for that. But maybe not in demo.
> > 
> > 
> > Yes, okay. Originally, I thought the rest of the worlds would be greyed out so it wouldn't matter which image we used ??
> 
> They will be but you still see the igloo and all, you can tell it's the same as ice world. Which is misleading.

Yeah I just downloaded it and I see what you mean. But I really like how @husseinfahmy  did the greying out. And the carousel looks really good with more than worlds. Is it possible for us to use an image just without the igloo. @natashadecoste Had a version of the world without the igloo and volcano. I think we can use that.

**Comment by rampras3** [comment]
> Regarding your first point - I didn't want to hide Natasha's pretty graphic hahaha. Maybe we can change the locked worlds to say coming soon? That way we can show the ice world while keeping it locked.

Personally, I like how it is right now. We should just remove the igloo in the image

**Comment by husseinfahmy** [comment]
Yea we can change the images easily! the locked world icon is just something I put together quickly for the sake of showing the alien/water/jungle worlds.

**Comment by rampras3** [comment]
@asmaaaah @husseinfahmy should we open another issue for the modal? It still redirects to another page at this point.

**Comment by rampras3** [comment]
> Yea we can change the images easily! the locked world icon is just something I put together quickly for the sake of showing the alien/water/jungle worlds.

Yeah, the lock icon looks good

**Comment by husseinfahmy** [comment]
I'm still working on the modal. Just trying to figure out how I'll get the the button (that's inside the carousel compoent) to show the modal (that's on the screen, outside the carousel). Seems like a basic solution but I'm useless with react native

**Comment by rampras3** [comment]
@husseinfahmy Can you add an instruction text that says "swipe to see the other worlds!" So they know to swipe

**Comment by natashadecoste** [comment]
Personally i think we should merge this PR and deal with any of the style fixes in new PRs (as a sort of 2nd iteration on top of our MVP sprint)
But i do agree we need to indicate somehow that the ice world is locked so i can probably gray it out slightly and then upload a new png to the repo or something (in a diff PR)

**Comment by asmaaaah** [comment]
@rampras3 if @natashadecoste other pictures then sure. I surely looks nice but it should not ruin the point that each world picture represents the world theme.

I am not asking to get rid of it completely. But just comment it out for now.

**Comment by rampras3** [comment]
> Personally i think we should merge this PR and deal with any of the style fixes in new PRs (as a sort of 2nd iteration on top of our MVP sprint)
> But i do agree we need to indicate somehow that the ice world is locked so i can probably gray it out slightly and then upload a new png to the repo or something (in a diff PR)

@husseinfahmy i think you should add the purple lock icon on the ice world as well to convey that it is locked

**Comment by husseinfahmy** [comment]
> @husseinfahmy Can you add an instruction text that says "swipe to see the other worlds!" So they know to swipe

How about I show the other worlds in the background like this:
image

**Comment by asmaaaah** [comment]
@husseinfahmy yes I agree with army on the lock:

1 . Lets but a lock to the Ice world
2. Let's comment out rest of the world for now
3. Yes showing the pictures in the background looks good!

**Comment by rampras3** [comment]
> I'm still working on the modal. Just trying to figure out how I'll get the the button (that's inside the carousel compoent) to show the modal (that's on the screen, outside the carousel). Seems like a basic solution but I'm useless with react native

The button in the carousel component can trigger a function call on click  this function can call the other function toggleEditState() -- this you can copy from the profile.js file. You can also copy the bottomsheet code in Profile.js but instead of all the stuff I put in there it would jsut be 2 buttons 
Inside carousel component: 
Your fire button should look like:
`                            <Button title="" onPress={() => this.toggleEditState()} />
`
This is outside of the carousel component, just in your render function of your screen:
```
                <BottomSheet onDismiss={() => this.toggleEditState()} visible={editMode} height={modalHeight}>
        <View>
                            <TouchableOpacity style={globalStyles.primaryBtn} onPress={() => this.displayAvatar()} >
                            <Text style={globalStyles.primaryBtnText}>Save</Text>
                            </TouchableOpacity>
                        </View>

                            <TouchableOpacity style={globalStyles.primaryBtn} onPress={() => this.cancelSave()} >
                            <Text style={globalStyles.primaryBtnText}>Cancel</Text>
                            </TouchableOpacity>
                    </View>
    </BottomSheet>

```


This basically, make the sheet only work for the Fire button for now.

File: src/components/WorldsConstants.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
+import * as DataObject from "./Database";
+import variables from '../styles/variables';
+
+export default WorldsConstants = {
```
Comment (aniprasad) [inline src/components/WorldsConstants.js:4]: While I like the fact that the constants are all in one file, I would prefer it they are moved into Constants.js as opposed to a new Constants file

Comment (husseinfahmy) [reply src/components/WorldsConstants.js:4]: Hmm, why not separate any constants that aren't related into their own file? Constants hierarchy shouldn't be unlike component hierarchy, no need to group them together if they're not referenced outside the worlds component.

I'm open to other points of view, that's just how I see it

**Pull request merged**